src/node_contextify.cc

```diff
@@ -270,6 +271,83 @@ void ContextifyContext::IsContext(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void ContextifyContext::GetWrappedFunction(
+    const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Local<Context> context = env->context();
+  Isolate* isolate = env->isolate();
+
+  CHECK(args[0]->IsString());
+  Local<String> code = args[0].As<String>();
+
+  CHECK(args[1]->IsBoolean());
+  bool produce_cached_data = args[1]->IsTrue();
+
+  Local<Uint8Array> cached_data_buf;
+  if (!args[2]->IsUndefined()) {
+    CHECK(args[2]->IsUint8Array());
+    cached_data_buf = args[2].As<Uint8Array>();
+  }
+
+  // TODO(ryzokuken): Add support for sandboxing.
+
+  ScriptCompiler::CachedData* cached_data = nullptr;
+  if (!cached_data_buf.IsEmpty()) {
+    ArrayBuffer::Contents contents = cached_data_buf->Buffer()->GetContents();
+    uint8_t* data = static_cast<uint8_t*>(contents.Data());
+    cached_data = new ScriptCompiler::CachedData(
+      data + cached_data_buf->ByteOffset(), cached_data_buf->ByteLength());
+  }
+
+  ScriptCompiler::Source source(code, cached_data);
+
+  ScriptCompiler::CompileOptions options;
+  if (source.GetCachedData() == nullptr) {
+    options = ScriptCompiler::kNoCompileOptions;
+  } else {
+    options = ScriptCompiler::kConsumeCodeCache;
+  }
+
+  Local<String> params[] = {
+    env->module_parameter_exports(),
+    env->module_parameter_require(),
+    env->module_parameter_module(),
+    env->module_parameter_filename(),
+    env->module_parameter_dirname()
+  };
+
+  TryCatch try_catch(isolate);
+  Context::Scope scope(context);
+
+  MaybeLocal<Function> maybe_fun = ScriptCompiler::CompileFunctionInContext(
+      context, &source, 5, params, 0, nullptr, options);
+
+  Local<Function> fun;
+  if (maybe_fun.IsEmpty() || !maybe_fun.ToLocal(&fun)) {
+    try_catch.ReThrow();
+    return;
+  }
+
+  if (produce_cached_data) {
+    const ScriptCompiler::CachedData* cached_data =
+      ScriptCompiler::CreateCodeCacheForFunction(fun, code);
+    bool cached_data_produced = cached_data != nullptr;
+    if (cached_data_produced) {
+      MaybeLocal<Object> buf = Buffer::Copy(
+          env,
+          reinterpret_cast<const char*>(cached_data->data),
+          cached_data->length);
+      fun->Set(env->cached_data_string(), buf.ToLocalChecked());
```

Comment:
@addaleax do you have a suggestions as to where the resulting buffer could be better stored?

Alternatively, we don't offer this at all, and expose a separate binding for `ScriptCompiler::CreateCodeCacheForFunction` similar to how we also recently [exposed](https://github.com/nodejs/node/pull/20300) `ScriptCompiler::CreateCodeCache`. That way this function gets a bit simpler, but then it would also lose symmetry to `ContextifyScript::New`.

Reply:
since the option for ContextifyScript::New is deprecated in my pr i think it makes sense to leave out the cache stuff for now.

Reply:
@devsnek could you drop me a link to it? Currently it `Set`s the two on the returned function object itself, which sounds like similar (if not the same) to what we're doing in `ContextifyScript::New`. Would love to see any alternatives you have in mind.

Reply:
https://github.com/nodejs/node/pull/20300

Reply:
@devsnek cool, this is exactly what @hashseed pointed out, I could do this. That said, it's still just a documentation-level deprecation, right?

Reply:
well... it will be docs-dep for 10, runtime dep for 11, and removed in 12.

Reply:
In that case it would be reasonable to not introduce it here in the first place :)